### PR TITLE
Am/chore/tfhe fft

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Review by mayeul-zama — COMMENTED** [review]
Rust code structure and `Cargo.toml` changes look fine to me

File: Makefile [hunk 1]
```diff
@@ -1289,6 +1298,116 @@ check_compile_tests
...
+		--all-features -p tfhe-fft -- --no-deps -D warnings
+
+.PHONY: pcc_fft # pcc stands for pre commit checks
+pcc_fft: check_fmt lint_doc_fft clippy_fft
+
+.PHONY: build_fft
+build_fft: install_rs_build_toolchain
+	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) build --release -p tfhe-fft
+	RUSTFLAGS="$(RUSTFLAGS)" cargo $(CARGO_RS_BUILD_TOOLCHAIN) build --release -p tfhe-fft \
+		--features=fft128
```
Comment (soonum) [inline Makefile:1336]: Wouldn't the second build overwrite the first ?
I'm curious about the change of build command paradigm comparing to `concrete-fft`, could you explain a bit more?

File: Makefile [hunk 2]
```diff
@@ -1289,6 +1298,116 @@ check_compile_tests
...
+test_fft_node_js_ci
+
+##### Bench #####
+
+.PHONY: bench_fft # Run FFT benchmarks
+bench_fft: install_rs_check_toolchain
+	RUSTFLAGS="$(RUSTFLAGS)" cargo "$(CARGO_RS_CHECK_TOOLCHAIN)" bench --bench fft -p tfhe-fft \
+		--features=serde \
+		--features=nightly \
+		--features=fft128
```
Comment (soonum) [inline Makefile:1408]: We force all the features by default ?

File: ci/fft_benchmark_parser.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
+"""
+benchmark_parser
+----------------
+
+Parse criterion benchmark.
```
Comment (soonum) [inline ci/fft_benchmark_parser.py:5]: Maybe add more context like:
```suggestion
Parse criterion benchmark for tfhe-fft.
```

File: ci/fft_benchmark_parser.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
+"""
+benchmark_parser
+----------------
```
Comment (soonum) [inline ci/fft_benchmark_parser.py:3]: Title to change to reflect filename.

File: .github/workflows/benchmark_tfhe_fft.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+          echo "Computing HMac on downloaded artifact"
+          SIGNATURE="$(slab/scripts/hmac_calculator.sh ${{ env.RESULTS_FILENAME }} '${{ secrets.JOB_SECRET }}')"
+          echo "Sending results to Slab..."
+          curl -v -k \
+          -H "Content-Type: application/json" \
+          -H "X-Slab-Repository: ${{ github.repository }}" \
+          -H "X-Slab-Command: store_data_v2" \
+          -H "X-Hub-Signature-256: sha256=${SIGNATURE}" \
+          -d @${{ env.RESULTS_FILENAME }} \
+          ${{ secrets.SLAB_URL }}
```
[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/benchmark_tfhe_fft.yml:128]: Wait, don't do that now. The database to send data to is currently hardcoded in Slab script :disappointed:

File: tfhe-fft/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,57 @@
+[package]
+name = "tfhe-fft"
+version = "0.6.0"
+edition = "2021"
+authors = ["sarah el kazdadi <sarah.elkazdadi@zama.ai>"]
```
Comment (soonum) [inline tfhe-fft/Cargo.toml:5]: Should we keep Sarah's name ? Or at least change the email address.

File: .github/workflows/benchmark_tfhe_fft.yml [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+          {
+            echo "BENCH_DATE=$(date --iso-8601=seconds)";
+            echo "COMMIT_DATE=$(git --no-pager show -s --format=%cd --date=iso8601-strict ${{ github.sha }})";
+            echo "COMMIT_HASH=$(git describe --tags --dirty)";
+          } >> "${GITHUB_ENV}"
+
+      - name: Install build dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y m4
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_tfhe_fft.yml:69]: huh, I thought this was done already, I need to remove it

File: .github/workflows/benchmark_tfhe_fft.yml [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  RESULTS_FILENAME: parsed_benchmark_results_${{ github.sha }}.json
+  RUST_BACKTRACE: "full"
+  SLACK_CHANNEL: ${{ secrets.SLACK_CHANNEL }}
+  SLACK_ICON: https://pbs.twimg.com/profile_images/1274014582265298945/OjBKP9kn_400x400.png
+  SLACK_USERNAME: ${{ secrets.BOT_USERNAME }}
+  SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+
+on:
+  workflow_dispatch:
+  push:
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_tfhe_fft.yml:16]: that's fine with me initially, but if you think it's not worth it I can change that, do you have an example of path in some of our workflows ?

File: .github/workflows/benchmark_tfhe_fft.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/benchmark_tfhe_fft.yml:128]: ok not taking this change

File: .github/workflows/cargo_build_tfhe_fft.yml [hunk 9]
```diff
@@ -0,0 +1,47 @@
+# Build tfhe-fft
+name: Cargo Build tfhe-fft
+
+on:
+  push:
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/cargo_build_tfhe_fft.yml:5]: no we want the check to be required, I don't think that works with the path ?

File: .github/workflows/make_release_tfhe_fft.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    inputs:
+      dry_run:
+        description: "Dry-run"
+        type: boolean
+        default: true
+
+env:
+  ACTION_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+jobs:
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/make_release_tfhe_fft.yml:15]: indeed

File: Makefile (hunk 2 above)
Comment (IceTDrinker) [reply Makefile:1408]: yes same logic as for the avx512, too many ways to make mistakes and we only care about the complete benchs

File: ci/fft_benchmark_parser.py (hunk 3 above)
Comment (IceTDrinker) [reply ci/fft_benchmark_parser.py:5]: yep

File: tfhe-fft/Cargo.toml (hunk 6 above)
Comment (IceTDrinker) [reply tfhe-fft/Cargo.toml:5]: We could change it same as in the TFHE-rs toml, I'll do that then

Comment (IceTDrinker) [reply tfhe-fft/Cargo.toml:5]: no authors lol

**Comment by IceTDrinker** [comment]
if it's fine with you I would like to go ahead without the paths/should run on the fft workflows for now and do it later to see how it behaves for a while, plus those only use github runners which should be fin on public repos

**Comment by IceTDrinker** [comment]
@soonum is it fine with you to continue with FFT tests at all times for now ? I have the follow up issue to reduce the test and bench frequency, though the tests are not very long and use some very small github runners

**Review by soonum — APPROVED** [review]
Go for it. We'll come back to fine tune workflows later.

**Pull request merged**